crates/bevy_pbr/src/render/utils.wgsl

```diff
@@ -17,8 +18,36 @@ fn hsv2rgb(hue: f32, saturation: f32, value: f32) -> vec3<f32> {
     return value * mix(vec3<f32>(1.0), rgb, vec3<f32>(saturation));
 }
 
-fn random1D(s: f32) -> f32 {
-    return fract(sin(s * 12.9898) * 43758.5453123);
+// Generates a random u32 in range [0, u32::MAX].
+//
+// `state` is a mutable reference to a u32 used as the seed.
+//
+// Values are generated via "white noise", with no correlation between values.
+// In shaders, you often want spatial and/or temporal correlation. Use a different RNG method for these use cases.
+//
+// https://www.pcg-random.org
+// https://www.reedbeta.com/blog/hash-functions-for-gpu-rendering
+fn rand_u(state: ptr<function, u32>) -> u32 {
+    *state = *state * 747796405u + 2891336453u;
+    let word = ((*state >> ((*state >> 28u) + 4u)) ^ *state) * 277803737u;
+    return (word >> 22u) ^ word;
+}
+
+// Generates a random f32 in range [0, 1.0].
+fn rand_f(state: ptr<function, u32>) -> f32 {
```

Comment:
I suppose this doesn't call into `rand_u` to avoid potential overhead with doing so?

Reply:
More like I just didn't think of doing that, but yeah might as well just keep it as-is.